pandas/core/generic.py

```diff
@@ -2909,6 +2909,9 @@ def _drop_axis(self, labels, axis, level=None, errors='raise'):
             else:
                 indexer = ~axis.isin(labels)
 
+            if errors == 'raise' and indexer.all():
+                raise ValueError('{} not found in axis'.format(labels))
```

Comment:
this should be a ``KeyError``

Reply:
Should line 3770 in pandas.core.indexes.base also be `KeyError` then?

Reply:
can u show a user facing example

Reply:
```
pd.DataFrame(index=['a', 'b']).drop('c')
ValueError: labels ['c'] not contained in axis
```

Reply:
if u change that,
what breaks?

Reply:
It breaks 15 tests that are expecting that scenario to be a `ValueError`

Reply:
can u show some more on this eg which tests

Reply:
```
============================================================= FAILURES =============================================================
_______________________________________________________ TestPanel.test_drop ________________________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/test_panel.py:2305: in test_drop
    pytest.raises(ValueError, panel.drop, 'Three')
pandas/core/generic.py:2863: in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
pandas/core/generic.py:2895: in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['Three'] not contained in axis"
____________________________________________ TestDataFrameSelectReindex.test_drop_names ____________________________________________
[gw1] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/frame/test_axis_select_reindex.py:44: in test_drop_names
    pytest.raises(ValueError, df.drop, ['g'])
pandas/core/generic.py:2863: in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
pandas/core/generic.py:2895: in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['g'] not contained in axis"
_______________________________________________ TestDataFrameSelectReindex.test_drop _______________________________________________
[gw0] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/frame/test_axis_select_reindex.py:90: in test_drop
    pytest.raises(ValueError, simple.drop, 5)
pandas/core/generic.py:2863: in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
pandas/core/generic.py:2895: in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: 'labels [5] not contained in axis'
_______________________________________________________ TestIndex.test_drop ________________________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1399: in test_drop
    pytest.raises(ValueError, self.strIndex.drop, ['foo', 'bar'])
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['foo' 'bar'] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop0-values0] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['a'] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop0-values1] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['a'] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop0-values2] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels ['a'] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop1-values0] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels [('c', 'd')] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop1-values1] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels [('c', 'd')] not contained in axis"
___________________________________________ TestIndex.test_drop_tuple[to_drop1-values2] ____________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/indexes/test_base.py:1451: in test_drop_tuple
    pytest.raises(ValueError, removed.drop, drop_me)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: "labels [('c', 'd')] not contained in axis"
_______________________________________________ TestPivotTable.test_pivot_no_values ________________________________________________
[gw3] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/reshape/test_pivot.py:214: in test_pivot_no_values
    res = df.pivot_table(index=df.index.month, columns=df.index.day)
pandas/core/frame.py:4513: in pivot_table
    margins_name=margins_name)
pandas/core/reshape/pivot.py:77: in pivot_table
    values = values.drop(key)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: 'labels [1 2 1 1 1] not contained in axis'
____________________________________________________ TestPivotTable.test_daily _____________________________________________________
[gw3] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/reshape/test_pivot.py:916: in test_daily
    columns=ts.index.dayofyear)
pandas/core/reshape/pivot.py:77: in pivot_table
    values = values.drop(key)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: 'labels [2000 2000 2000 ..., 2004 2004 2004] not contained in axis'
___________________________________________________ TestPivotTable.test_monthly ____________________________________________________
[gw3] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/reshape/test_pivot.py:934: in test_monthly
    columns=ts.index.month)
pandas/core/reshape/pivot.py:77: in pivot_table
    values = values.drop(key)
pandas/core/indexes/base.py:3771: in drop
    labels[mask])
E   KeyError: 'labels [2000 2000 2000 2000 2000 2000 2000 2000 2000 2000 2000 2000 2001 2001 2001\n 2001 2001 2001 2001 2001 2001 2001 2001 2001 2002 2002 2002 2002 2002 2002\n 2002 2002 2002 2002 2002 2002 2003 2003 2003 2003 2003 2003 2003 2003 2003\n 2003 2003 2003 2004 2004 2004 2004 2004 2004 2004 2004 2004 2004 2004 2004] not contained in axis'
___________________________________________________ TestSeriesIndexing.test_drop ___________________________________________________
[gw2] darwin -- Python 3.5.4 /Users/Alex/miniconda3/envs/pandas/bin/python
pandas/tests/series/test_indexing.py:1841: in test_drop
    pytest.raises(ValueError, s.drop, 'bc')
pandas/core/generic.py:2863: in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
pandas/core/generic.py:2895: in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
pandas/core/indexes/base.py:3771: in drop
    'labels %s not contained in axis' % labels[mask])
E   KeyError: "labels ['bc'] not contained in axis"
```

Reply:
yeah let me have a closer look
we could prob just change this to be a KeyError which is more consistent with other indexing

Reply:
Great, thanks!